src/middlewared/middlewared/schema.py

```diff
@@ -138,9 +139,11 @@ def clean(self, value):
         if value is None:
             return value
         if isinstance(value, int) and not isinstance(value, bool):
-            return str(value)
+            value = str(value)
         if not isinstance(value, str):
             raise Error(self.name, 'Not a string')
+        if not self.empty and not value:
```

Comment:
Wont this be a problem with Int when value = 0?

Reply:
It is only for `Str` and `List`

Reply:
Ah, the diff fooled me, I thought this was inside Attribute ; )